### DISCUSSION TITLE
Setting up Tina CMS with NextJS and Netlify is throwing a lot of errors

### DISCUSSION THREAD

**Discussion by Isaac-Tait** [tinacms/tinacms#4049] [Help] [answered]
[Here](https://github.com/Isaac-Tait/royalRidgesPaintball) is my repo. I am working on getting my NextJS/TinaCMS project to build on Netlify. However, when I attempt to build on Netlify I get this error:

```
9:00:20 AM: $ yarn build
9:00:21 AM: yarn run v1.22.19
9:00:21 AM: $ tinacms build && next build
9:00:22 AM: Starting Tina build
9:00:24 AM: {
9:00:24 AM:   heading: Error when checking client information,
9:00:24 AM:   items: [
9:00:24 AM:     {
9:00:24 AM:       emoji: ❌,
9:00:24 AM:       heading: You provided,
9:00:24 AM:       subItems: [
9:00:24 AM:         {
9:00:24 AM:           key: clientId,
9:00:24 AM:           value: ****
9:00:24 AM:         },
9:00:24 AM:         {
9:00:24 AM:           key: token,
9:00:24 AM:           value: ****
9:00:24 AM:         }
9:00:24 AM:       ]
9:00:24 AM:     }
9:00:24 AM:   ]
9:00:24 AM: }
9:00:24 AM: Error: Server responded with status code 500, Internal Server Error.
9:00:24 AM: Message from server: Something went wrong. Please see our FAQ for more information: 
```

When I run `next build` locally I get this error:

```
> Build error occurred
Error: Failed to collect page data for /demo/blog/[filename]
    at /usr/local/lib/node_modules/next/dist/build/utils.js:916:15
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  type: 'Error'
}
info  - Collecting page data .%
```

When I run `yarn build` I get this error:

```
yarn run v1.22.19
$ tinacms build && next build
Starting Tina build
│
○  Error when checking client information ──────────────────╮
│                                                           │
│  ❌ You provided                                           │
│     clientId:   myClientID     │
│     token:      myToken  │
│                                                           │
│                                                           │
├───────────────────────────────────────────────────────────╯
Error: Server responded with status code 500, Internal Server Error.

Message from server: Something went wrong. Please see our FAQ for more information: https://tina.io/docs/errors/faq/
```

The dichotomy in error messages is making it pretty hard for me to track down what is wrong.

**Comment by kldavis4** [comment]
Looks like you have the client id configured as `myClientID` and token as `myToken`

**Reply by Isaac-Tait** [reply]
I changed that just for this discussion - the error message shows my full un-obfuscated ClientID and Token. I am assuming that I need to keep those private, is that correct?

**Reply by kldavis4** [reply]
only the token needs to be protected. Does your project start correctly when you run the dev command? I can also check the logs on the server if you provide the client id.

**Reply by Isaac-Tait** [reply]
Thank you 😄 - my client ID is 99157794-e14d-4477-8da0-eed38046dcda

My project does start locally correctly when I run `yarn dev`

**Reply by kldavis4** [reply, marked as answer]
It looks like the token is bad. Maybe there is whitespace in it? I forked your project and set it up in tina cloud and was able to run the build without any issues.

<img width="1236" alt="Screen Shot 2023-06-28 at 10 03 52" src="https://github.com/tinacms/tinacms/assets/370955/735d1117-abad-40de-aedd-3f0bbd42842c">

**Reply by kldavis4** [reply]
@Isaac-Tait was it a whitespace issue?

**Reply by Isaac-Tait** [reply]
Ha! I wish it was... 🤣 I was using the search token not the READ ONLY one 🤦🏻‍♂️😬

**Reply by Isaac-Tait** [reply]
Thank you for your help and sleuthing out my mistake 👍🏼
